core/src/dbs/session.rs

```diff
@@ -27,6 +28,8 @@ pub struct Session {
 	pub tk: Option<Value>,
 	/// The current scope authentication data
 	pub sd: Option<Value>,
+	/// The current expiration time of the session
+	pub exp: Option<i64>,
```

Comment:
I understand we are using i64 instead of std time because the local clock can change. What about i64 vs u64?

Reply:
I think a `u64` would strictly speaking make more sense here as well. The choice of the type was made to match the `exp` claim from the token, which is [parsed as `i64`](https://github.com/surrealdb/surrealdb/blob/e06cd111cf7ac005f80f9ee85c2fd6f742f689ea/core/src/iam/token.rs#L18) and is the actual origin of the session expiration value. It is not an issue that the value can be signed, but I see how `u64` is more clean if doing so does not add any unneeded complexity. I will look into changing this :+1:

Reply:
After looking into this, I think I prefer to leave it as is. The reason is that converting the `exp` claim to `u64` would need to be done gracefully and either return an error or just create a session expiring at epoch zero, which would be expired. If this will end up being the result anyway, I prefer session expiration to be handled the same as token expiration, which requires allowing a negative expiration time but rejecting any expired token. Likewise, in sessions we would be allow sessions to exist (and be created) with negative expiration times, but the session will be already expired and thus prevented from doing anything which requires an authenticated session.